### ISSUE TITLE
yolo8 pose train ignoring corrupt image/label: labels require 56 columns each

### ISSUE THREAD

**Issue by WANGCHAO1996** [ultralytics/ultralytics#3041]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

when i train pose on coco valdata  the warning occur:ignoring corrupt image/label: labels require 56 columns each how to resolve it .?

**Comment by Laughing-q** [comment]
@WANGCHAO1996 seems like you got incorrect label format, each row in pose label of coco set needs 56 elements, cls+box+keypoints(1+4+17*3=56). And the warning says you don't have exact 56 elements for each row.

**Comment by WANGCHAO1996** [comment]
> @WANGCHAO1996 seems like you got incorrect label format, each row in pose label of coco set needs 56 elements, cls+box+keypoints(1+4+17*3=56). And the warning says you don't have exact 56 elements for each row.

![image](https://github.com/ultralytics/ultralytics/assets/51087601/d3a1c085-10ea-4656-a495-b6bd09d58c12)
Yes, that's the problem. However, I downloaded the coco val pose data set and used yolov8s-pose to train why the accuracy would decrease and could not reach the accuracy index given in the project?

**Comment by glenn-jocher** [comment]
@WANGCHAO1996, thank you for informing us about the issue with your pose labels. You need to ensure that each row in your pose label of COCO dataset has a length of 56, which includes the cls, box, and 17 keypoints each with x and y coordinates (1+4+17*2=56). The error message that you are seeing indicates that there are rows in your dataset that don't have 56 columns and hence, can not be read. 

Regarding the accuracy, note that YOLOv8 is an extension of YOLOv5 and not YOLOv5, so please ensure you have followed the recommended training tips for YOLOv8 for the pose-detection task. You can find these tips in the YOLOv8 documentation. 

Hope this helps! Please don't hesitate to ask if you run into any further issues.

**Comment by WANGCHAO1996** [comment]
> @WANGCHAO1996, thank you for informing us about the issue with your pose labels. You need to ensure that each row in your pose label of COCO dataset has a length of 56, which includes the cls, box, and 17 keypoints each with x and y coordinates (1+4+17*2=56). The error message that you are seeing indicates that there are rows in your dataset that don't have 56 columns and hence, can not be read.
> 
> Regarding the accuracy, note that YOLOv8 is an extension of YOLOv5 and not YOLOv5, so please ensure you have followed the recommended training tips for YOLOv8 for the pose-detection task. You can find these tips in the YOLOv8 documentation.
> 
> Hope this helps! Please don't hesitate to ask if you run into any further issues.

thank you! i used this conmmand : yolo task=pose mode=train  model= yolov8s-pose.pt  data=/data/work/wangch/ultralytics-main/ultralytics/datasets/coco-pose.yaml  batch=16 epochs=100 imgsz=640 workers=4 device=\'0,1\' 

Is there anything else you should be aware of？

**Comment by WANGCHAO1996** [comment]
> > @WANGCHAO1996, thank you for informing us about the issue with your pose labels. You need to ensure that each row in your pose label of COCO dataset has a length of 56, which includes the cls, box, and 17 keypoints each with x and y coordinates (1+4+17*2=56). The error message that you are seeing indicates that there are rows in your dataset that don't have 56 columns and hence, can not be read.
> > Regarding the accuracy, note that YOLOv8 is an extension of YOLOv5 and not YOLOv5, so please ensure you have followed the recommended training tips for YOLOv8 for the pose-detection task. You can find these tips in the YOLOv8 documentation.
> > Hope this helps! Please don't hesitate to ask if you run into any further issues.
> 
> thank you! i used this conmmand : yolo task=pose mode=train model= yolov8s-pose.pt data=/data/work/wangch/ultralytics-main/ultralytics/datasets/coco-pose.yaml batch=16 epochs=100 imgsz=640 workers=4 device='0,1'
> 
> Is there anything else you should be aware of？

use this?
from ultralytics import YOLO

# Load a model
model = YOLO('yolov8n-pose.yaml')  # build a new model from YAML
model = YOLO('yolov8n-pose.pt')  # load a pretrained model (recommended for training)
model = YOLO('yolov8n-pose.yaml').load('yolov8n-pose.pt')  # build from YAML and transfer weights

# Train the model
model.train(data='coco8-pose.yaml', epochs=100, imgsz=640)

**Comment by glenn-jocher** [comment]
@WANGCHAO1996, it seems like you have followed the correct command to train your pose detection task with YOLOv8. Just make sure that your dataset labels have a fixed length of 56 which includes the cls, box, and 17 keypoints each with x and y coordinates (1+4+17*2=56). 

Regarding your question, the answer is no. You don't need to use the provided code that you have posted. The training command you are currently using is enough to train your model. Please note that YOLOv8 offers various configuration files and training tips specifically designed for different tasks. You can find more information in the YOLOv8 documentation.

Let me know if you have any further questions or if there is anything else I can do to help.

**Comment by Cornflowerzbq** [comment]
Hello, have you resolved it? My dataset has 8 keypoints, it should be 1+4+8*2, but the code is giving an error saying "labels require 29 columns each".

**Comment by HonestyBrave** [comment]
i have a question, if one picture have 4 same point label other picture have 6 same point label, the point label name is same

then the  kpt_shape: [X, 3] should set what, X=4 or X=6?

**Comment by pderrenger** [comment]
The `kpt_shape` parameter should be set to the maximum number of keypoints that any image in your dataset has. In your case, if one image has 4 keypoints and another has 6, you should set `kpt_shape: [6, 3]` to accommodate the maximum number of keypoints. This ensures that all keypoints are accounted for during training.
